### PR TITLE
feat: deprecate `important` property in favor of `addToSummaryDashboard` in CustomMetricGroup

### LINKED ISSUE #242: `add_to_summary_dashboard` does not work on `monitor_custom`
### Version

1.22.4

### Steps and/or minimal code example to reproduce

I'm using the python version:
1. Change dashboard factory to also create the summary
```
dashboard_factory=monitoring.DefaultDashboardFactory(
                self,
                "DashboardFactory",
                dashboard_name_prefix="streamProcessors",
                create_dashboard=True,
                detail_dashboard_range=aws_cdk.Duration.hours(4),
            ),
```
2. monitor a custom resource and supply `add_to_summary_dashboard=True`
```
monitoring_obj.monitor_custom(
            alarm_friendly_name="Processor performance",
            add_to_summary_dashboard=True,
            metric_groups=[]
)
```

### Expected behavior

Expected to see the custom monitoring widget in the summary like other resources (dynamo DB in this case) I set to be shown in the summary.

### Actual behavior

`monitor_custom` contents are not shown in the summary.

### Other details

_No response_

### PR DESCRIPTION
**Pull request by Glyphack**
Resolves #242 

To keep the add to summary naming consistent the new property has been added to CustomMetricGroup.

---

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license_

### CODE REVIEW

File: lib/monitoring/custom/CustomMonitoring.ts [hunk 1]
```diff
@@ -304,7 +310,10 @@ export class CustomMonitoring extends Monitoring {
   private getAllWidgets(summary: boolean): IWidget[] {
     const filteredMetricGroups = summary
       ? this.metricGroups.filter(
-          (group) => group.metricGroup.important ?? false
+          (group) =>
+            group.metricGroup.important ??
+            group.metricGroup.addToSummaryDashboard ??
```
Comment (echeung-amzn) [inline lib/monitoring/custom/CustomMonitoring.ts:315]: `addToSummaryDashboard` should probably take precedence over `important` if present. Should have tests around that too.

**Review by echeung-amzn — CHANGES_REQUESTED** [review]

File: lib/monitoring/custom/CustomMonitoring.ts (hunk 1 above)
Comment (Glyphack) [reply lib/monitoring/custom/CustomMonitoring.ts:315]: It makes sense.
Regarding the tests, do you mean adding a new test for case `important: false, addToSummaryDashboard: true` right?

Comment (echeung-amzn) [reply lib/monitoring/custom/CustomMonitoring.ts:314]: > Regarding the tests, do you mean adding a new test for case `important: false, addToSummaryDashboard: true` right?

Yup.

**Review by echeung-amzn — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**